connpicker.go

```diff
@@ -0,0 +1,87 @@
+package gocql
+
+import (
+	"sync"
+	"sync/atomic"
+)
+
+type ConnPicker interface {
+	Pick(token) *Conn
+	Put(*Conn)
+	Remove(conn *Conn)
+	Size() (int, int)
+	Close()
+}
+
+type DefaultConnPicker struct {
+	conns []*Conn
+	pos   uint32
+	size  int
```

Comment:
I'd use an unsigned int variant for size.

Reply:
It's very viral. This property is threaded through the code and it makes sense to have them as `uint` but perhaps not in this patch?